src/app/recipes/recipe-edit/recipe-edit.component.html

```diff
@@ -1,8 +1,64 @@
-<h1>Recipe edit</h1>
-<hr>
-<div *ngIf="recipe">
-    <h1>name: {{recipe.name}}</h1>
-    <h2>description: {{recipe.description}}</h2>
-    <h3>image path: {{recipe.imagePath}}</h3>
-    <h3>id: {{recipe.id}}</h3>
+<div class="row">
+    <div class="col-xs-12">
+        <form [formGroup]="recipeForm" (ngSubmit)="onSubmit()">
+            <div class="row">
+                <div class="col-xs-12">
+                    <button type="submit" class="btn btn-success" [disabled]="!recipeForm.valid">Save</button>
+                    <button type="button" class="btn btn-danger" (click)="onCancel()">Cancel</button>
+                </div>
+            </div>
+            <div class="row">
+                <div class="col-xs-12">
+                    <div class="form-group">
+                        <label for="name">Name</label>
+                        <input type="text" id="name" class="form-control" formControlName="name">
+                    </div>
+                </div>
+            </div>
+            <div class="row">
+                <div class="col-xs-12">
+                    <div class="form-group">
+                        <label for="imagePath">Image URL</label>
+                        <input type="text" id="imagePath" #imagePath class="form-control" formControlName="imagePath">
+                    </div>
+                </div>
+            </div>
+            <div class="row">
+                <div class="col-xs-12">
+                    <img [src]="imagePath.value" class="img-responsive">
+                </div>
+            </div>
+            <div class="row">
+                <div class="col-xs-12">
+                    <div class="form-group">
+                        <label for="description">Image URL</label>
+                        <textarea type="text" id="description" class="form-control" rows="6"
+                            formControlName="description"></textarea>
+                    </div>
+                </div>
+            </div>
+            <div class="row">
+                <div class="col-xs-12" formArrayName="ingredients">
+                    <div class="row" 
+                    *ngFor="let ingredientsCtrl of (recipeForm.get('ingredients')).controls; let i = index"
```

Comment:
Where do you use ingredientsCtrl ? Is controller of Ingredients or just ingredientControl?

Reply:
It is just a ingredientControl. Used to render an ingredient list.